include/marl/pool.h

```diff
@@ -214,32 +212,64 @@ class BoundedPool : public Pool<T> {
 
   // borrow() borrows a single item from the pool, blocking until an item is
   // returned if the pool is empty.
-  MARL_NO_EXPORT inline Loan borrow() const;
+  template <typename... Args>
+  MARL_NO_EXPORT inline Loan borrow(Args&&... args) const {
```

Comment:
It would be less accident prone if we could ensure that there are no arguments when `POLICY == PoolPolicy::Preserve`. Either through a `std::enable_if`, or even a `static_assert`.

Reply:
done